python/ray/data/context.py

```diff
@@ -227,6 +227,11 @@ def __init__(
         # The additonal ray remote args that should be added to
         # the task-pool-based data tasks.
         self._task_pool_data_task_remote_args: Dict[str, Any] = {}
+        # Max number of task failures that are allowed before aborting
+        # the Dataset execution. Data of the failed tasks will be dropped.
+        # Unlimited if negative.
+        # By default, not failures are allowed.
+        self.max_allowed_task_failures = 0
```

Comment:
For consistency with the `DataContext` options, should we expose this in the constructor and define a constant to represent the default?

Reply:
I thought of that. But now the parameter list is already too long, and in practice we never create the DataContext objects with parameters. Thus, I prefer to not add new parameters.